### ISSUE TITLE
Segmentation fault - cannot use GUI

### ISSUE THREAD

**Issue by Jnorth01** [Ettercap/ettercap#1254]
Hopefully someone can assist me with this.  Ettercap crashes in GUI after hitting the accept button.  debug log is attached.  Maybe has something to do with GTK? Also during build, i get this warning: 

The package name passed to `find_package_handle_standard_args` (GTK3_GTK)  
  does not match the name of the calling package (GTK3).  This can lead to   
  problems in calling code that expects `find_package` result variables      
  (e.g., `_FOUND`) to follow a certain pattern.         

[ettercap-0.8.4-rc_debug.log](https://github.com/user-attachments/files/15523213/ettercap-0.8.4-rc_debug.log)

Using a new install of Kali - only thing I have done was install the wifi adapter driver and upgraded the system with apt-get upgrade.

Any advice is appreciated.

**Comment by koeppea** [comment]
Could you please as well upload the core file so that I am able to restore the state of ettercap's memory at the time if the crash?

**Comment by Jnorth01** [comment]
Thank you for the quick reply.  It took me a while to figure out how to get the core file to generate.  Evidently the package systemd-coredump was missing.
[core.ettercap.zip](https://github.com/user-attachments/files/15526869/core.ettercap.zip)

**Comment by TimsPizza** [comment]
Same problem here. Same situation: 
1. Updated the system with apt upgrade. Linux version now is 
`Linux null 6.6.15-amd64 #1 SMP PREEMPT_DYNAMIC Kali 6.6.15-2kali1 (2024-05-17) x86_64 GNU/Linux`
2. Found the 88XXau driver is corrupted. Removed all related drivers and then deleting any related files in this path
`/lib/modules/$(uname -r)/kernel/drivers/net/wireless/`
3. Recompiled aircrack-ng/rtl8812au and installed with dkms. WLAN returned to normal.
4. Found ettercap crashing: seg fault.

GDB debugging information:

`
(gdb) r -G
Starting program: /usr/local/bin/ettercap -G
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

ettercap 0.8.4-rc copyright 2001-2020 Ettercap Development Team

[New Thread 0x155551dff6c0 (LWP 90448)]
[New Thread 0x155551bfe6c0 (LWP 90449)]
[Detaching after fork from child process 90450]
[New Thread 0x1555519fd6c0 (LWP 90451)]
[New Thread 0x1555507ff6c0 (LWP 90452)]
[New Thread 0x15555022a6c0 (LWP 90477)]
[Detaching after fork from child process 90479]
[New Thread 0x1555409ff6c0 (LWP 90480)]
[New Thread 0x1555407fe6c0 (LWP 90481)]
[New Thread 0x1555405fd6c0 (LWP 90482)]
[New Thread 0x1555403fc6c0 (LWP 90483)]
[Thread 0x1555403fc6c0 (LWP 90483) exited]

Thread 1 "ettercap" received signal SIGSEGV, Segmentation fault.
0x0000155554695ec9 in ?? () from /lib/x86_64-linux-gnu/libgio-2.0.so.0
(gdb) bt
#0  0x0000155554695ec9 in  () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#1  0x00001555546968b1 in  () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#2  0x00001555546bab58 in g_dbus_connection_register_object () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#3  0x0000155554696fa1 in g_dbus_connection_export_menu_model () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#4  0x00001555540117b5 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#5  0x0000155553cbc82f in gtk_application_set_app_menu () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#6  0x0000155555460bc4 in gtkui_create_menu (app=0x555555c59870, data=0x1) at /home/kali/ettercap/src/interfaces/gtk3/ec_gtk3_menus.c:504
#7  0x00001555547ab730 in g_closure_invoke () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#8  0x00001555547bf87c in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#9  0x00001555547c1281 in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#10 0x00001555547c6f06 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#11 0x00001555547c6fc3 in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#12 0x000015555468a290 in  () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#13 0x000015555468a426 in g_application_run () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#14 0x000015555545a356 in gtkui_start () at /home/kali/ettercap/src/interfaces/gtk3/ec_gtk3.c:945
#15 0x00001555554cd226 in ui_start () at /home/kali/ettercap/src/ec_ui.c:63
#16 0x00001555554b414d in libettercap_ui_start () at /home/kali/ettercap/src/ec_libettercap.c:60
#17 0x0000555555558ab7 in main (argc=2, argv=0x7fffffffdd78) at /home/kali/ettercap/src/ec_main.c:201
`

**Comment by Jnorth01** [comment]
I decided to reinstall Kali (using the installer image from their site) and Ettercap worked as I expected.  I then repeated the steps I did initially (apt update, apt upgrade, apt dist-upgrade) but without installing the WiFi adapter and Ettercap started crashing again.  I reinstalled Kali for the 3rd time, installed the WiFi adapters without upgrading the system and Ettercap is still running.  I guess I will stay away from upgrading the system for now.
